### ISSUE TITLE
is there a standardised way to interact with LLMs

### ISSUE THREAD

**Issue by mosajjal** [iaalm/llama-api-server#33]
Hi. 

has anyone come up with a unified, standard API to talk to all LLMs? (LLaMA, Bard, OpenAI etc)? if not, it would be helpful to start defining that as a community and build the compatibility layers for that. keen to hear your thoughts.

**Comment by iaalm** [comment]
OpenAI API is the de facto standard in this area. That's why I start this project to call custom LLM in OpenAI way. Any suggestion?

**Comment by mosajjal** [comment]
I agree that currently OpenAI is considered the standard. However, I don't think OpenAI is committed to keep the API specification intact for the foreseeable future.

My suggestion is to start a community-led group to codify a standard, and start building a compatibility shim between major models (OpenAI, LLaMA etc) according to said model. We can always start with OpenAI's reference and build on top of it.

**Comment by iaalm** [comment]
Agree, that would be very useful😎

**Comment by mosajjal** [comment]
cool :) so how should we start this? need to start a group of some sort and start exchanging ideas on what it would look like.

**Comment by mosajjal** [comment]
sounds good. I can help with Bard.
